app/controllers/api/v1/student_quizzes_controller.rb

```diff
@@ -0,0 +1,86 @@
+class Api::V1::StudentQuizzesController < ApplicationController
+  include AuthorizationHelper
+
+  # Check if the current user is authorized to perform the action
+  def action_allowed?
+    if current_user_is_a? 'Student'
+      if action_name.eql? 'index'
+        are_needed_authorizations_present?(params[:id], 'reviewer', 'submitter')
+      else
+        true
+      end
+    else
+      current_user_has_ta_privileges? && user_ta_for_current_course?
+    end
+  end
+
+  # Fetches and lists all quiz response mappings for a specific participant in an assignment.
+  def index
+    @participant = AssignmentParticipant.find(params[:id])
+    # Ensures the current user is authorized to view the participant's quiz mappings.
+    return unless current_user_id?(@participant.user_id)
+
+    @assignment = Assignment.find(@participant.parent_id)
+    @quiz_mappings = ResponseMap.mappings_for_reviewer(@participant.id)
+  end
+
+  # Displays the questions and participant responses for a completed quiz
+  def show_quiz_responses
+    @response = Response.find(params[:response_id])
+    @response_map = ResponseMap.find(params[:map_id])
+    @questions = Question.where(questionnaire_id: @response_map.reviewed_object_id)
+    @participant = AssignmentTeam.find(@response_map.reviewee_id).participants.first
+    @quiz_score = @response.aggregate_questionnaire_score # Use the score calculated by Response model
+  end
+
+  # Fetches quizzes for a given assignment that a reviewer has not yet taken.
+  def fetch_available_quizzes_for_reviewer(assignment_id, reviewer_id)
+    quizzes = []
+    reviewer = Participant.find_by(user_id: reviewer_id, parent_id: assignment_id)
+    review_response_maps = ReviewResponseMap.where(reviewer_id: reviewer.id)
+
+    review_response_maps.each do |response_map|
+      reviewee_team = Team.find(response_map.reviewee_id)
+
+      next unless reviewee_team.parent_id == assignment_id
```

Comment:
Why do we need to cycle through all the response  maps?  Couldn't we just do a find to find the response map with the right parent?